setup.py

```diff
@@ -3,6 +3,27 @@
 import versioneer
 
 
+using_ipy4 = False
+try:
+    from IPython import __version__ as ipyv
+    from distutils.version import LooseVersion
+
+    using_ipy4 = (ipyv >= LooseVersion("4"))
+except ImportError:
+    using_ipy4 = True
+
+install_requires = [
+    'Jinja2>=2.7.2',
+    'Pygments>=1.6',
+    'ipython>=2.3.1',
+    'pyzmq>=14.1.0',
+]
+if using_ipy4:
+    install_requires.extend([
+        'ipykernel>=4.0.0',
+        'nbformat>=4.0.0',
```

Comment:
It is worth noting that we did not end up requiring `notebook`. Only a few very basic `jupyter` dependencies were installed (e.g. `jupyter_core`, `jupyter_client`, `ipython_genutils`, `traitlets`, and some other python or ipython related-ish packages).